cumulusci/core/config/marketing_cloud_service_config.py

```diff
@@ -47,8 +44,11 @@ def _refresh_token(self):
         * access_token
         * refresh_token
         """
-        oauth_client = OAuth2Client(self._client_config.config)
-        info = oauth_client.refresh_token(self.refresh_token)
+        oauth2_client_config = self._keychain.get_service(
+            "oauth2_client", self.oauth2_client
+        )
+        oauth2_client = OAuth2Client(oauth2_client_config.config)
```

Comment:
@davisagli - Can we just pass the instance of `OAuth2CLientConfig` instead of it's config dict?

Reply:
@Br4nd0R great point; adjusted